single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/layout/pathfinding/DijkstraPathFinder.java

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.sld.layout.pathfinding;
+
+import java.util.*;
+
+/**
+ * @author Thomas Adam <tadam at neverhack.com>
+ */
+public final class DijkstraPathFinder extends AbstractPathFinder {
+
+    private static final int[] DX = {1, 0, -1, 0};  // Horizontal moves
+    private static final int[] DY = {0, 1, 0, -1};  // Vertical moves
+
+    public DijkstraPathFinder() {
+        // Nothing to do
+    }
+
+    @Override
+    public List<Point> findShortestPath(Grid grid, int startX, int startY, int endX, int endY, boolean setSnakeLineAsObstacle) {
+        Point start = new Point(startX, startY);
+        Point end = new Point(endX, endY);
+        Map<Point, Integer> distance = new HashMap<>();
+
+        PriorityQueue<Point> queue = new PriorityQueue<>(Comparator.comparingInt(distance::get));
+
+        distance.put(start, 0);
+        queue.add(start);
+
+        while (!queue.isEmpty()) {
```

Comment:
Could we have `queue.isEmpty()` equal to `true`?

Reply:
If no neighbors available no point will be added to `queue` and no path will be found